src/Features/VisualBasic/Portable/GenerateConstructor/VisualBasicGenerateConstructorService.vb

```diff
@@ -242,6 +242,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.GenerateConstructor
                     End If
                 End If
 
+                If newNode.Kind = SyntaxKind.AsNewClause Then
+                    ' Speculation is not supported for AsNewClauseSyntax nodes.
+                    ' Generate an EqualsValueSyntax node with the inner NewExpression of the AsNewClauseSyntax node for speculation.
+
+                    Dim asNewClauseNode = DirectCast(newNode, AsNewClauseSyntax)
+                    Dim equalsValueNode = SyntaxFactory.EqualsValue(asNewClauseNode.NewExpression)
+                    newNode = asNewClauseNode.CopyAnnotationsTo(equalsValueNode)
```

Comment:
Is this necessary? The node you're looking for on the next line is a _child_ of equalsValueNode/newNode that's already annotated with `s_annotation`. This line would copy any annotations from the higher-level AsNew to the EqualsValue, but I don't see where that matters downstream.